arch/arm/src/common/Toolchain.defs

```diff
@@ -47,7 +47,11 @@ endif
 ifeq ($(CONFIG_FRAME_POINTER),y)
   ARCHOPTIMIZATION += -fno-omit-frame-pointer -fno-optimize-sibling-calls
 else
-  ARCHOPTIMIZATION += -fomit-frame-pointer
+  ifeq ($(CONFIG_ARM_TOOLCHAIN_GHS),y)
+    ARCHOPTIMIZATION += -noga
```

Comment:
Seriously I don't like adding too many special compiler assertions to the common toolchain. For tricore architecture, GNUC/TASKING is separated into two files, because most of the compilation options are completely different:
https://github.com/apache/nuttx/blob/master/arch/tricore/src/common/ToolchainTasking.defs
https://github.com/apache/nuttx/blob/master/arch/tricore/src/common/ToolchainGnuc.defs

Reply:
> Seriously I don't like adding too many special compiler assertions to the common toolchain. For tricore architecture, GNUC/TASKING is separated into two files, because most of the compilation options are completely different: https://github.com/apache/nuttx/blob/master/arch/tricore/src/common/ToolchainTasking.defs https://github.com/apache/nuttx/blob/master/arch/tricore/src/common/ToolchainGnuc.defs

Ok, I'm still working on it, and will split the toolchain specific implementation do isolate files